### PR TITLE
Simplify broken limb mending

### PR DESCRIPTION
**Pull request by Coolthulhu**
#### Summary

SUMMARY: Balance "Simplify broken limb mending"

#### Purpose of change

Make broken limb not take you out of combat for [insert way too long duration here].

#### Describe the solution

Broken limbs can now be healed in every way a healthy limb can, they just heal at 25% speed, unless splinted or you have REGEN_LIZ.
Limbs stay broken until they reach full hp.

#### Describe alternatives you've considered

* Reworking limb breaking to be purely hp based, so that limb is broken when below x% hp
* Keeping the long wait until we have proper limb removal

#### Testing

* Set hp of limbs to, respectively: 0, 0, 0, 50%
* Set one of the 0 hp limbs to 50% hp
* Splint one of the 0 hp limbs
* The splinted limb should now be "blue" on the hp bar, two broken ones should be red
* Sleep until the splinted limb visibly regens about 25%
* Splinted limb should regen as much as the non-broken limb
* The non-splinted limbs should heal only about 1/4 of that amount
* Set hp of all broken limbs to full
* Note they unbreak on the next turn
* Set leg hp to 0, 0, then up to 50%, 50%, then wait a turn
* Try to run - should fail
* Set leg hp to 100%, 100%, then wait a turn
* Try to run - should succeed
* Set arm hp to 0, 100%, then 50%, 100%
* Try to fire a bow - should fail
* Set arm hp to 100%, 100%, then wait a turn
* Try to fire a bow - should succeed

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Does limb fixing machine taken into account to the change?

**Comment by chaosvolt** [comment]
Question about this: does this obsolete `mending_modifier` as a mutation property? I feel like retaining it as a modifier for mending that's separate from regular healing would still be reasonable, among other things it wouldn't render the liz mending trait worthless.

**Comment by Coolthulhu** [comment]
> I mean Auodoc bonesetting.

It sets your hp to 50% if it's lower and the limb is broken.

> Question about this: does this obsolete mending_modifier as a mutation property?

At the moment yes. I should probably fix that and at least obsolete it properly.

> It'd make more sense to use mending modifier there instead, and presumably accompany this by changing lizard regen's mending modifier to 4.0 so that it achieves the same effect of "cancel out the 0.25 healing modifier if you're a gecko"

Could do. I picked the hardcode route mostly to avoid "break your limb to heal it faster" meta that the 20x heal would have, but I guess I could cap it at full speed or something.

**Comment by Coolthulhu** [comment]
Alright, changed to respect the `mending_modifier`, although had to redefine it.

File: src/character.h [hunk 1]
```diff
@@ -742,10 +742,8 @@ class Character : public Creature, public visitable<Character>
         int get_working_arm_count() const;
         /** Returns the number of functioning legs */
         int get_working_leg_count() const;
-        /** Returns true if the limb is disabled(12.5% or less hp)*/
+        /** Returns true if the limb is disabled(12.5% or less hp and not broken)*/
```
Comment (olanti-p) [inline src/character.h:745]: Probably meant the opposite here

File: src/iexamine.cpp [hunk 2]
```diff
@@ -5166,11 +5165,8 @@ void iexamine::autodoc( player &p, const tripoint &examp )
             for( int i = 0; i < num_hp_parts; i++ ) {
                 const bodypart_id &part = convert_bp( player::hp_to_bp( static_cast<hp_part>( i ) ) ).id();
                 const bool broken = patient.is_limb_broken( part );
-                effect &existing_effect = patient.get_effect( effect_mending, part->token );
                 // Skip part if not broken or already healed 50%
```
Comment (olanti-p) [inline src/iexamine.cpp:5168]: Outdated comment

**Review by olanti-p — COMMENTED** [review]
Code looks good, can reproduce tests.
Description for the `disabled` effect might need an update.

Would be nice if there was a more explicit indicator or a message that the split works, since less experienced people might miss the color change in the sidebar.

**Review by olanti-p — APPROVED** [review]

**Pull request merged**